File: Examples/CSharp/Working-with-Objects/BuildTangentAndBinormalData.cs
using System;
using System.IO;
using System.Collections;
using Aspose.ThreeD;
using Aspose.ThreeD.Animation;
using Aspose.ThreeD.Entities;
using Aspose.ThreeD.Formats;

namespace Aspose._3D.Examples.CSharp._Working_with_Objects
{
    class BuildTangentAndBinormalData
    {
        public static void Run()
        {
            // ExStart:BuildTangentAndBinormalData
            // Load an existing 3D file
            Scene scene = new Scene(RunExamples.GetDataFilePath("document.fbx"));
            // Triangulate a scene
            PolygonModifier.BuildTangentBinormal(scene);
            // Save 3D scene
            scene.Save(RunExamples.GetOutputFilePath("BuildTangentAndBinormalData_out.fbx"), FileFormat.FBX7400ASCII);
            // ExEnd:BuildTangentAndBinormalData              
        }
    }
}

File: Examples/VisualBasic/Working-with-Objects/BuildTangentAndBinormalData.vb
Imports System.IO
Imports System.Collections
Imports Aspose.ThreeD
Imports Aspose.ThreeD.Animation
Imports Aspose.ThreeD.Entities
Imports Aspose.ThreeD.Formats

Namespace Working_with_Objects
    Class BuildTangentAndBinormalData
        Public Shared Sub Run()
            ' ExStart:BuildTangentAndBinormalData
            ' The path to the documents directory.
            Dim MyDir As String = RunExamples.GetDataDir()

            ' Load an existing 3D file
            Dim scene As New Scene(MyDir & Convert.ToString("document.fbx"))
            ' Triangulate a scene
            PolygonModifier.BuildTangentBinormal(scene)
            ' Save 3D scene
            scene.Save("BuildTangentAndBinormalData_out.fbx", FileFormat.FBX7400ASCII)
            ' ExEnd:BuildTangentAndBinormalData              
        End Sub
    End Class
End Namespace

